dungeon/build.gradle

```diff
@@ -71,15 +71,39 @@ tasks.register('runRoomBasedDungeon', JavaExec) {
 }
 
 
-// create our Starter.jar
+// Generate license report
+import com.github.jk1.license.render.*
+
+licenseReport {
+    unionParentPomLicenses = true
+
+    outputDir = "${rootProject.project('dungeon').buildDir}/licenses"
+
+    // For game and dungeon dependencies (default is 'all projects'):
+    projects = [rootProject.project('game'), rootProject.project('dungeon')]
```

Comment:
Guten Morgen @cagix,

ich habe die Syntax hier noch nicht richtig verstanden, glaube ich.

Es macht einen Unterschied, ob ich `projects = rootProject.project('game'), rootProject.project('dungeon')` oder `projects = rootProject.project('dungeon'), rootProject.project('game')` schreibe, und das finde ich merkwürdig. Liegt das vielleicht auch an `unionParentPomLicenses`?

Leider ist diese Option auch nicht sehr gut beschrieben.

Kannst du bitte noch einmal selber dir diese Stelle anschauen? Danke

Reply:
@tgrothe Es würde wie immer helfen, wenn Du nicht einfach nur schreiben würdest "Ugh, Problem!", sondern wenn Du Deine Beobachtungen zusammenfassen könntest. Was hast Du gesehen, wo gibt es Unterschiede, was wäre aus Deiner Sicht korrekt, was ist nicht wirklich richtig, ... Dann kann ich auch Zeit dafür aufbringen und weiter nachschauen. Aber so kann ich damit nix anfangen, weil ich ja erstmal verstehen muss, was Du da gesehen hast und was Dir wohl seltsam vorkommt.

Reply:
@cagix Danke für die Antwort, ich hab mir das mal genauer angesehen.

```
//unionParentPomLicenses = true
//projects = [rootProject.project('game'), rootProject.project('dungeon')]
1.md

//unionParentPomLicenses = true
projects = rootProject.project('game')
2.md

//unionParentPomLicenses = true
projects = rootProject.project('dungeon')
3.md

//unionParentPomLicenses = true
projects = rootProject.project('game') rootProject.project('dungeon')
4.md

//unionParentPomLicenses = true
projects = rootProject.project('dungeon') rootProject.project('game')
5.md

//unionParentPomLicenses = true
projects = [rootProject.project('game'), rootProject.project('dungeon')]
6.md

//unionParentPomLicenses = true
projects = [rootProject.project('dungeon'), rootProject.project('game')]
7.md

unionParentPomLicenses = true
projects = [rootProject.project('game'), rootProject.project('dungeon')]
8.md

$ for ((i=1; i<=8; ++i)); do for ((j=i+1; j<=8; ++j)); do diff -I '^_.*' -q "$i.md" "$j.md"; done; done
Files 1.md and 2.md differ
Files 1.md and 5.md differ
Files 2.md and 3.md differ
Files 2.md and 4.md differ
Files 2.md and 6.md differ
Files 2.md and 7.md differ
Files 2.md and 8.md differ
Files 3.md and 5.md differ
Files 4.md and 5.md differ
Files 5.md and 6.md differ
Files 5.md and 7.md differ
Files 5.md and 8.md differ

```

Das bedeutet nun:

- Die Option `unionParentPomLicenses` hat bei uns keinen Effekt
- Nur `projects = rootProject.project('game')` oder nur `projects = rootProject.project('dungeon')` macht einen Unterschied
- `projects = rootProject.project('game') rootProject.project('dungeon')` oder `projects = rootProject.project('dungeon') rootProject.project('game')` greift jeweils nur das erste Argument auf
- `projects = [rootProject.project('game'), rootProject.project('dungeon')]` und `projects = [rootProject.project('dungeon'), rootProject.project('game')]` sind identisch
- `projects = [rootProject.project('game'), rootProject.project('dungeon')]` wäre richtig - also genau so, wie es zurzeit ist

-> resolved

Reply:
ok - also war es ein syntaxproblem? die schreibweise `projects = A B` ist nicht gültig, sondern es muss `projects = [A, B]` heissen? danke fürs nachprüfen!

Reply:
ich gehe mal davon aus, dass du den code entsprechend nachgezogen hast?

Reply:
@tgrothe deine betrachtungen passen aber nur, wenn der default von `unionParentPomLicenses` tatsächlich `false` ist. hast du das mal explizit auf `false` gesetzt für 1..7?

zweite frage: du betrachtest in deinem kommentar nur einen diff der dateien, ohne auf die inhalte einzugehen. hattest du geprüft, welche der dateien das erwartete ergebnis beinhalten, also welche der varianten inhaltlich korrekt ist? (das kann ich an deiner antwort nicht ersehen.)

Reply:
@cagix ich habe die reports jetzt einmal zum vergleich mit eingecheckt -> b19ffa24c2475f09e9ee24af2f22b7e52526c8fa

wenn du sagst, das geht so, dann nehme ich die reports wieder raus